nncf/quantization/algorithms/fast_bias_correction/algorithm.py

```diff
@@ -118,50 +125,49 @@ def _apply(self,
 
         nncf_graph = NNCFGraphFactory.create(model)
         node_and_bias_value = (
-            (node, self._backend_entity.get_bias_value(node, nncf_graph, model)) \
-                for node in nncf_graph.get_all_nodes() if self._backend_entity.is_node_with_bias(node, nncf_graph)
+            (node, self._backend_entity.get_bias_value(node, nncf_graph, model))
+            for node in nncf_graph.get_all_nodes()
+            if self._backend_entity.is_node_with_bias(node, nncf_graph, model)
         )
+
         model_transformer = ModelTransformerFactory.create(model)
         # Fill `node_and_new_bias_value` list. It is a correspondence between nodes
         # for which we should update bias and new bias values.
         node_and_new_bias_value = []
         for node, bias_value in node_and_bias_value:
             node_name = node.node_name
 
-            if not self._backend_entity.is_quantized_weights(node, nncf_graph):
+            if not self._backend_entity.is_quantized_weights(node, nncf_graph, model):
                 nncf_logger.debug(f'Skipping node {node_name} because weights were not quantized')
                 continue
 
             input_fp, input_shape = self._get_fp_inputs(statistic_points, node_name)
             output_fp = self._get_fp_outputs(statistic_points, node_name)
 
-
-            extracted_model = self._extract_submodel(model_transformer,
-                                                     node_name)
+            extracted_model = self._extract_submodel(model_transformer, node_name)
 
             sub_input_name, sub_output_name = self._backend_entity.get_sub_input_output_names(extracted_model)
 
             channel_axis = node.metatype.output_channel_axis
             if bias_value.ndim > 1:
                 # Make index positive
                 channel_axis = range(bias_value.ndim)[channel_axis]
-            input_blob = self._create_input_data(input_shape,
-                                                 input_fp,
-                                                 sub_input_name,
-                                                 channel_axis)
+            input_blob = self._create_input_data(input_shape, input_fp, sub_input_name, channel_axis)
             bias_shift = self._get_bias_shift(
                 model=extracted_model,
                 input_blob=input_blob,
                 channel_axis=channel_axis,
                 output_fp=output_fp,
-                output_name=sub_output_name)
+                output_name=sub_output_name
+            )
 
             if bias_value.ndim > 1:
-                axes = [i for i in range(bias_value.ndim) if i != channel_axis]
-                bias_shift = np.expand_dims(bias_shift, axes)
+                new_shape = [1] * bias_value.ndim
+                new_shape[channel_axis] = bias_shift.shape[0]
+                bias_shift = bias_shift.reshape(new_shape)
```

Comment:
I would make a specific function that could be tested if needed. One more good point is that it will have typehints with all possible types of `bias_shift`. It is not explicit now

Reply:
But it will be 3 functions absolutely identity expect of typehint. 
But np.array typehint, better change to 'TTensor' like used `TModel`.

Reply:
I mean one common function

Reply:
How you suggest to add `typehints with all possible types of bias_shift` in one common function?

Reply:
`typehints with all possible types of bias_shift` - it was my misunderstanding. Now I see that this is can't be done in the common code.

Reply:
Moved to function, added test